runtime/queries/metricsview_aggregation.go

```diff
@@ -847,6 +847,9 @@ func (q *MetricsViewAggregation) buildTimestampExpr(mv *runtimev1.MetricsViewSpe
 			return fmt.Sprintf("date_trunc('%s', %s)", dialect.ConvertToDateTruncSpecifier(dim.TimeGrain), col), nil, nil
 		}
 		return fmt.Sprintf("toTimezone(date_trunc('%s', toTimezone(%s::TIMESTAMP, ?)), ?)", dialect.ConvertToDateTruncSpecifier(dim.TimeGrain), col), []any{dim.TimeZone, dim.TimeZone}, nil
+	case drivers.DialectPinot:
+		// ToDateTime format truncates millis to secs because we don't support that, for example timeseries api does timestamppb.New(ts) which truncates to seconds
+		return fmt.Sprintf("ToDateTime(date_trunc('%s', %s, 'MILLISECONDS', '%s'), 'yyyy-MM-dd''T''HH:mm:ss''Z''')", dialect.ConvertToDateTruncSpecifier(dim.TimeGrain), col, dim.TimeZone), nil, nil
```

Comment:
yes datetrunc will return epoch millis, so ideally the timestamp format here should match the timestamp format returned from timeseries call because front end does exact match while rendering timeseries while comparing dimensions

Reply:
unfortunately the return type of `date_trunc` func is `LONG` not `TIMESTAMP`